test/unit_test/character_sheet_unit_tests/FieldRulesTests.java

```diff
@@ -117,23 +122,39 @@ public void testGetIntegerByField_ArmorPenalty() {
 		assertEquals(expected, actual);
 	}
 
-	// @Test
-	// public void testGetBoolByField_True() {
-	// final FieldRules fieldRules = new FieldRules(TEST_RULE_FILE_PATH);
-	// final boolean actual =
-	// fieldRules.<Boolean>getValueByField(Background.ANDER_SURVIVOR,
-	// "boolDataTrue");
-	// assertTrue(actual);
-	// }
-	//
-	// @Test
-	// public void testGetBoolByField_False() {
-	// final FieldRules fieldRules = new FieldRules(TEST_RULE_FILE_PATH);
-	// final boolean actual =
-	// fieldRules.<Boolean>getValueByField(Background.ANDER_SURVIVOR,
-	// "boolDataFalse");
-	// assertFalse(actual);
-	// }
+	@Test
+	public void testGetAndOrEnumsByField_EmptyList() {
+		final FieldRules fieldRules = new FieldRules(TEST_RULE_FILE_PATH);
+		HashSet<HashSet<Focuses>> expecteds = new HashSet<>();
+		HashSet<HashSet<Focuses>> actuals = fieldRules.getAndOrEnumsForField(Talents.ARMOR_TRAINING, Focuses.class, Keys.FOCUSES);
+		assertEquals(expecteds, actuals);
+	}
+
+	@Test
+	public void testGetAndOrEnumsByField_FocusesORRelation() {
+		final FieldRules fieldRules = new FieldRules(TEST_RULE_FILE_PATH);
+		HashSet<HashSet<Focuses>> expecteds = new HashSet<>();
+		HashSet<Focuses> focusesList = new HashSet<>();
+		focusesList.add(Focuses.PERFORMANCE);
+		focusesList.add(Focuses.MUSICAL_LORE);
+		expecteds.add(focusesList);
+		HashSet<HashSet<Focuses>> actuals = fieldRules.getAndOrEnumsForField(Talents.MUSIC, Focuses.class, Keys.FOCUSES);
+		assertEquals(expecteds, actuals);
+	}
+
+	@Test
+	public void testGetAndOrEnumsByField_FocusesANDRelation() {
+		final FieldRules fieldRules = new FieldRules(TEST_RULE_FILE_PATH);
+		HashSet<HashSet<Focuses>> expecteds = new HashSet<>();
+		HashSet<Focuses> focusesList1 = new HashSet<>();
+		HashSet<Focuses> focusesList2 = new HashSet<>();
+		focusesList1.add(Focuses.LEGERDEMAIN);
+		focusesList2.add(Focuses.STEALTH);
+		expecteds.add(focusesList1);
+		expecteds.add(focusesList2);
+		HashSet<HashSet<Focuses>> actuals = fieldRules.getAndOrEnumsForField(SpecializationClasses.SHADOW, Focuses.class, Keys.FOCUSES);
+		assertEquals(expecteds, actuals);
+	}
```

Comment:
Ez és a felette levő teszt ugyan olyan értékekkel operál. Látom, hogy eltérő logikai operátor művelet lenne a cél, de számomra ez az eltérés a teszt kódsorából, persze a nevét leszámítva nem tükröződik.

Reply:
Nem a művelet tér el,  hanem az adat eltérő típusú, és ebből kifolyólag a funkció más peremkapabilitása van tesztelve. Ha érdekel elmagyarázom szóban, nyugodtan csörrents meg.

Reply:
Illetve ha megnézed a két expected is különböző. Hiszen az elsőben egy innerset van 2 elemmel. Míg ebben két singleton innerset van